ucb-rit/coldfront
# Replace majority of user-facing, hard-coded strings with settings

Fixes #384

**Changes**
- Introduced three new settings:
	- `PORTAL_NAME`: "MyBRC" or "MyLRC"
	- `PROGRAM_NAME_LONG`: "Berkeley Research Computing" or "Laboratory Research Computing"
	- `PROGRAM_NAME_SHORT`: "BRC" or "LRC"
- Introduced corresponding Ansible variables for configuring them in `main.yml`.
- Replaced hard-coded strings with the new settings.
- Added a context processor `portal_and_program_names` so that the new settings are available in templates without being explicitly set as context variables.
- Refactored LRC-/BRC-specific documentation text in emails sent when a user joins/is added to a project into separate templates.
- Refactored the feedback alert into a template.
- Corrected `full_host_path` in the development `main.copyme` to fix the nav-bar "Help" link.

**How to Test**
- Review the code changes and add comments as needed.
	- Search for various terms like "Berkeley", "BRC", "MyBRC", "CalNet", "Savio", etc. Note any instances not covered by the exceptions below.
- Ensure that the test suite passes.
- Manual Testing
	- On both an LRC-based VM and a BRC-based VM, checkout the branch, configure new Ansible variables, re-run the playbook, and look for inconsistencies or errors around the UI.

**Notes**
- Not all deployment-specific words have been replaced. Notable exceptions include:
	- The user access agreement
	- The survey for new project requests
	- Savio/Vector new project requests (including URL paths)
	- `ACCOUNT_CREATION_TEXT` and some login/registration/password-reset help text
		- These can be ignored because they should not appear based on flags introduced in `issue_276`.
	- Feedback alert
	- The "Savio Compute" `Resource`
